src/datachain/model/yolo.py

```diff
@@ -0,0 +1,500 @@
+    right_knee = 14
+    left_ankle = 15
+    right_ankle = 16
+
+
+class YoloPose(YoloBox):
+    """
+    A class representing human pose keypoints detected by a YOLO model.
+
+    This class extends the `YoloBox` class to include the segments
+    of the detected objects.
+
+    Pose estimation is a task that involves identifying the location of specific points
+    in an image, usually referred to as keypoints.
+
+    See https://docs.ultralytics.com/tasks/pose/ for more information.
+
+    Attributes:
+        cls (list[int]): A list of classes of the detected objects, default `[]`.
+        name (list[str]): A list of names of the detected objects, default `[]`.
+        confidence (list[float]): A list of confidence scores of the detections,
+            default `[]`.
+        box (list[list[float]]): A list of bounding boxes of the detected objects,
+            stored as pixels coordinates [x_min, y_min, x_max, y_max]
+            (PASCAL VOC format), default `[]`.
+        keypoints (list[list[list[float]]]): A list of human pose keypoints
+            of the detected objects, stored as a list of x and y coordinates
+            and visibility score, default `[]`.
+        orig_shape (list[int]): The original size of the image (height, width),
+            default `[]`.
+
+    Note:
+        There are 17 keypoints in total, each represented by a pair of x and y
+        coordinates and a visibility score. The keypoints can be accessed by name
+        using the `datachain.model.YoloPoseBodyPart` enumeration.
+    """
+
+    keypoints: list[list[list[float]]] = Field(default=[])
+
+    @staticmethod
+    def from_yolo_results(results: list["Results"]) -> "YoloPose":
+        """
+        Create a YOLO pose keypoints from the YOLO results.
+
+        Example:
+            ```python
+            from ultralytics import YOLO
+            from datachain.model.bbox import YoloPose
+
+            model = YOLO("yolo11n-pose.pt")
+            results = model("image.jpg", verbose=False)
+            segments = YoloPose.from_yolo_results(results)
+            ```
+
+        Args:
+            results: YOLO results from the model.
+
+        Returns:
+            YoloPose: A YOLO pose keypoints data model.
+        """
+        if not (summary := results[0].summary(normalize=False)):
+            return YoloPose()
+
+        cls, name, confidence, box, keypoints = [], [], [], [], []
+        for res in summary:
+            cls.append(res.get("class", -1))
+            name.append(res.get("name", ""))
+            confidence.append(res.get("confidence", -1))
+            box.append(_get_box_from_yolo_result(res))
+            keypoints.append(_get_keypoints_from_yolo_result(res))
+
+        return YoloPose(
+            cls=cls,
+            name=name,
+            confidence=confidence,
+            box=box,
+            keypoints=keypoints,
+            orig_shape=list(results[0].orig_shape),
+        )
+
+
+class YoloCls(DataModel):
```

Comment:
Not sure if anyone will use it but Yolo have models so I've added this model here too.